browser/extensions/api/brave_rewards_api.cc

```diff
@@ -1189,6 +1195,57 @@ BraveRewardsShouldShowOnboardingFunction::Run() {
   return RespondNow(OneArgument(base::Value(should_show)));
 }
 
+BraveRewardsGetScheduledCaptchaInfoFunction::
+    ~BraveRewardsGetScheduledCaptchaInfoFunction() = default;
+
+ExtensionFunction::ResponseAction
+BraveRewardsGetScheduledCaptchaInfoFunction::Run() {
+#if BUILDFLAG(BRAVE_ADAPTIVE_CAPTCHA_ENABLED)
+  Profile* profile = Profile::FromBrowserContext(browser_context());
+  auto* brave_adaptive_captcha_service =
+      brave_adaptive_captcha::BraveAdaptiveCaptchaServiceFactory::GetForProfile(
+          profile);
+  if (!brave_adaptive_captcha_service) {
+    return RespondNow(Error("Adaptive captcha service is not initialized"));
+  }
+
+  std::string url;
+  bool max_attempts_exceeded = false;
+  brave_adaptive_captcha_service->GetScheduledCaptchaInfo(
+      &url, &max_attempts_exceeded);
+#endif
+
+  base::DictionaryValue dict;
+  dict.SetString("url", url);
```

Comment:
this is not even going to build if BRAVE_ADAPTIVE_CAPTCHA_ENABLED is false

Reply:
the only reason you aren't getting a compile failure here is because this file is not included in ios.

Reply:
Thanks, fixed.